pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -349,7 +352,7 @@ protected void readMoreEntries(Consumer consumer) {
         }
     }
 
-    protected int calculateNumOfMessageToRead(Consumer consumer) {
+    protected MutablePair<Integer, Integer> calculateToRead(Consumer consumer) {
```

Comment:
Is it possible to move to `AbstractBaseDispatcher`? Looks many duplicated code with the PersistentDispatcherMultipleConsumers.